gmm_client/src/client.erl

```diff
@@ -1,106 +1,118 @@
-  {ok, Body} = client_requests:get_simple_request_body(?URL++"graph/vertices/listing"),
-  list_body_elements(binary:bin_to_list(Body)).
-
-% do ewentualnej serializacji
-get_vertex_info(Id)->
-  application:ensure_all_started(hackney),
-  Url= ?URL++"graph/vertices/details?id="++Id,
-  {ok, Body} = client_requests:get_delete_request(Url, get),
-  list_body_elements(binary:bin_to_list(Body)).
-
-
-delete_vertex(Id)->
-  application:ensure_all_started(hackney),
-  Url = ?URL++"graph/vertices/delete?id="++Id,
-  client_requests:post_request(Url).
-
-% EDGES
-
-add_edge(From, To, Permissions, Trace, Successive)->
-  application:ensure_all_started(hackney),
-  TraceString = case Trace of
-                  undefined -> "";
-                  _ -> "&trace="++Trace
-                end,
-  Url= ?URL++"graph/edges?from="++From++"&to="++To++"&permissions="++Permissions++TraceString++"&successive="++atom_to_list(Successive),
-  client_requests:post_request(list_to_binary(Url)).
-
-% do ewentualnej serializacji
-set_edge_permissions(From, To, Permissions, Trace, Successive)->
-  application:ensure_all_started(hackney),
-  Url= ?URL++"graph/edges/permissions?from="++From++"&to="++To++"&permissions="++Permissions++"&trace="++Trace++"&successive="++atom_to_list(Successive),
-  client_requests:post_request(list_to_binary(Url)).
-
-check_edge_existance(From, To)->
-  application:ensure_all_started(hackney),
-  Url= ?URL++"is_adjacent?from="++From++"&to="++To,
-  client_requests:post_request(list_to_binary(Url)).
-
-get_edge_permissions(From, To)->
-  application:ensure_all_started(hackney),
-  Url= ?URL++"permissions?from="++From++"&to="++To,
-  client_requests:post_request(list_to_binary(Url)).
-
-get_vertex_children(Of)->
-  application:ensure_all_started(hackney),
-  Url= ?URL++"list_adjacent_reversed?of="++Of,
-  client_requests:post_request(list_to_binary(Url)).
-
-get_vertex_parents(Of)->
-  application:ensure_all_started(hackney),
-  Url= ?URL++"list_adjacent?of="++Of,
-  client_requests:post_request(list_to_binary(Url)).
-
-delete_edge(From, To, Trace, Successive)->
-  application:ensure_all_started(hackney),
-  Url= ?URL++"graph/edges/delete?from="++From++"&to="++To++"&trace="++Trace++"&successive="++atom_to_list(Successive),
-  client_requests:post_request(list_to_binary(Url)).
+-export([
+    add_vertex/3,
+    get_all_vertices/1,
+    get_vertex/1,
+    delete_vertex/1,
+    add_edge/4,
+    set_permissions/4,
+    delete_edge/3,
+    edge_exists/2,
+    get_permissions/2,
+    list_parents/1,
+    list_children/1
+]).
+
+
+%%%---------------------------
+%% Implementations
+%%%---------------------------
+
+%% VERTICES
+
+-spec add_vertex(Zone :: binary(), Type :: binary(), Name :: binary()) -> {ok, binary()} | {error, any()}.
+add_vertex(Zone, Type, Name) ->
+    {ok, Address} = http_utils:get_address(Zone),
+    Url = http_utils:build_url(Address, <<"graph/vertices">>,
+        [{<<"type">>, Type}, {<<"name">>, Name}]),
```

Comment:
Z tego co widzę, bardzo podobna konstrukcja występuje dość często. Warto by zrobić z tego osobną funkcję

Reply:
występuje podobna konstrukcja, ale moim zdaniem takie rozbicie na kilka linijek jest bardziej czytelne -> budujemy sobie adres Url, a potem wykonujemy zapytanie. Jakby to wrzucić do wywołania jednej funkcji, to byłoby dość sporo argumentów

Reply:
Przekształcenie 3 linijek na 2 linijki nie jest moim zdaniem warte utraty czytelności i potencjalnej utraty/utrudnienia kontroli nad błędami